src/utils/nviHelpers.ts

```diff
@@ -8,3 +17,125 @@ export const getNviYearFilterValues = () => {
   }
   return nviYearFilterValues;
 };
+
+const areTopLevelOrgsEqual = <T>(set1: Set<T>, set2: Set<T>) => {
+  if (set1.size !== set2.size) {
+    return false;
+  }
+  for (const item of set1) {
+    if (!set2.has(item)) {
+      return false;
+    }
+  }
+  return true;
+};
+
+const hasChangedContributorsOrAffiliations = async (
+  persistedContributors: Contributor[],
+  updatedContributors: Contributor[]
+) => {
+  const topLevelOrgCache: { [key: string]: string } = {};
+
+  const getTopLevelOrgId = async (affiliationId: string) => {
+    if (!topLevelOrgCache[affiliationId]) {
+      const organization = await fetchOrganization(affiliationId);
+      const topLevelOrg = getTopLevelOrganization(organization);
+      topLevelOrgCache[affiliationId] = topLevelOrg.id;
+    }
+    return topLevelOrgCache[affiliationId];
+  };
+
```

Comment:
småting som _kan_ gjøre ting litt mer lesbart er å ta enkelting i en egen blokk, feks:
```suggestion
if (persistedContributors.length !== updatedContributors.length) {
  return true;
}
// fortsett med andre sjekker
```
da slipper man at alt annet kommer med i samme innrykk